[Docs] Removing paragraph about PDO similarity

ThomasLandauer opened the pull request:
According to https://www.doctrine-project.org/2020/11/17/dbal-3.0.0.html, this is not true anymore.

> The major theme of DBAL 3.0 is the decoupling from PDO.

ThomasLandauer committed:
Removing outdated references to PDO

morozov commented:
I believe it's not the only obsolete PDO-related statement. Here's another one:
https://github.com/doctrine/dbal/blob/44fb02006088c85dec9d6f4a93b642fc8395a8c0/README.md#L10

@ThomasLandauer could you look through the rest of the documentation and see if there are more?

ThomasLandauer commented:
OK, I guess I caught most of them now ;-)
This caution box at https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/transactions.html#transaction-nesting I don't understand, so I left it:
> Directly invoking PDO#beginTransaction(), PDO#commit() or PDO#rollBack() or the corresponding methods on the particular Doctrine\DBAL\Driver\Connection instance in use bypasses the transparent transaction nesting that is provided by Doctrine\DBAL\Connection and can therefore corrupt the nesting level, causing errors with broken transaction boundaries that may be hard to debug.

greg0ire commented:
I think that box only made sense on 2.x, where methods from `PDO` where callable through a `Driver\Connection` instance: https://github.com/doctrine/dbal/blob/01595c011c1aff35088a1f3ae6823222738c511d/lib/Doctrine/DBAL/Driver/PDOConnection.php#L23

On 3.x, inheritance has been removed and composition is used instead: https://github.com/doctrine/dbal/blob/44fb02006088c85dec9d6f4a93b642fc8395a8c0/src/Driver/PDO/Connection.php#L16-L19

EDIT: my bad, it still makes sense:
> Directly invoking PDO#beginTransaction(), PDO#commit() or PDO#rollBack()

You can use getWrappedConnection():

https://github.com/doctrine/dbal/blob/44fb02006088c85dec9d6f4a93b642fc8395a8c0/src/Driver/PDO/Connection.php#L147-L150

> or the corresponding methods

Or you can call those methods , they are proxied:

https://github.com/doctrine/dbal/blob/44fb02006088c85dec9d6f4a93b642fc8395a8c0/src/Driver/PDO/Connection.php#L126-L145

I think the warning can be left as is.

ThomasLandauer commented:
OK, thanks!

morozov commented on docs/en/reference/introduction.rst:
```diff
@@ -1,17 +1,15 @@
-existing native or self-made APIs. For example, the DBAL ships with
-a driver for Oracle databases that uses the oci8 extension under
-the hood.
+The Doctrine database abstraction & access layer (DBAL) offers an
```
This is the first time I see it unabbreviated as "abstraction & access". Let's keep it as a "database abstraction layer".

ThomasLandauer replied:
Me too :-)

morozov commented on docs/en/reference/security.rst:
```diff
@@ -90,7 +90,7 @@ Instead of using string concatenation to insert user-input into your SQL/DQL sta
 which placeholder. Each database vendor supports different placeholder styles:
 
--  All PDO Drivers support positional (using question marks) and named placeholders (:param1, :foo, :bar).
+-  All drivers support positional (using question marks) and named placeholders (e.g. ``:param1``, ``:foo``).
```
"All drivers support positional" — this is not true. The original statement was correct.

morozov commented on docs/en/reference/transactions.rst:
```diff
@@ -1,10 +1,10 @@
-``Connection#beginTransaction()``, ``Connection#commit()`` and
-``Connection#rollBack()``.
+``beginTransaction()``, ``commit()`` and
+``#rollBack()``.
```
The `#` looks redundant.

morozov commented:
@ThomasLandauer once the work is complete, please squash all changes in one commit.

ThomasLandauer commented:
Sorry, I don't know how to do that :-( Contrary to what @greg0ire recommended at https://github.com/doctrine/orm/pull/9160#issuecomment-961466986 I did it on GitHub's website again ;-)

greg0ire commented:
Here is some help once you do the dance described in https://github.com/doctrine/orm/pull/9160#issuecomment-961466986:

How to do that?

1. `git rebase -i origin/3.1.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

ThomasLandauer commented:
Here's what I'm getting after exiting nano :-(

> CONFLICT (modify/delete): lib/Doctrine/ORM/Persisters/OneToManyPersister.php deleted in HEAD and modified in e50f77a78... Corrected method names; the interface already used SQL, the files still used Sql in method names. Version e50f77a78... Corrected method names; the interface already used SQL, the files still used Sql in method names of lib/Doctrine/ORM/Persisters/OneToManyPersister.php left in tree.
CONFLICT (modify/delete): lib/Doctrine/ORM/Persisters/ManyToManyPersister.php deleted in HEAD and modified in e50f77a78... Corrected method names; the interface already used SQL, the files still used Sql in method names. Version e50f77a78... Corrected method names; the interface already used SQL, the files still used Sql in method names of lib/Doctrine/ORM/Persisters/ManyToManyPersister.php left in tree.
error: could not apply e50f77a78... Corrected method names; the interface already used SQL, the files still used Sql in method names
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply e50f77a78... Corrected method names; the interface already used SQL, the files still used Sql in method names

The first line in nano looked like the others - are you sure to keep "pick" there?

greg0ire commented:
I tried the rebase locally, and apparently it contains commits that are not yours:

```
p 8ddf46b60 [Raphael Stolt @ 11 days ago] Fixes spelling
p 5ed8c9b98 [Michael Voříšek @ 2 weeks ago] MariaDb1027Platform class should be non-final
p 07540a0b2 [Sergei Morozov @ 13 days ago] Update Psalm to 4.11.2
p 194448094 [Sergei Morozov @ 2 weeks ago] Rework negation of the special characters in SQL parser
p 9d38a4865 [Sergei Morozov @ 13 days ago] Handle PCRE error in the SQL parser
p a354bbb7a [Sergei Morozov @ 11 days ago] Workflow for closing stale pull requests
p 81c54e8ca [Sergei Morozov @ 7 days ago] Fix connection leaks in Oracle functional tests
p 317c05620 [Sergei Morozov @ 7 days ago] Use CONCAT() with SQL Server to concatenate strings
p 0ce5d7892 [Sergei Morozov @ 6 days ago] Do not mark issues as stale
p 3d2f79f2d [Thomas Landauer @ 27 hours ago] Removing paragraph about PDO similarity
p c3721d990 [Thomas Landauer @ 23 hours ago] Update README.md
p 2de7b5b5a [Thomas Landauer @ 23 hours ago] Update README.md
p 8a5d0173e [Thomas Landauer @ 23 hours ago] Update introduction.rst
p 6cf002580 [Thomas Landauer @ 23 hours ago] Update data-retrieval-and-manipulation.rst
p a6edade26 [Thomas Landauer @ 23 hours ago] Removing "This is a Doctrine 2.1 feature"
p c54601f54 [Thomas Landauer @ 23 hours ago] Update query-builder.rst
p 4732b690e [Thomas Landauer @ 23 hours ago] Update security.rst
p 363fbd6c5 [Thomas Landauer @ 23 hours ago] Update transactions.rst
p 5226d0112 [Thomas Landauer @ 22 hours ago] Update transactions.rst
p 3ca8ba648 [Thomas Landauer @ 9 hours ago] Update introduction.rst
p 9b6ac8f50 [Thomas Landauer @ 9 hours ago] Update security.rst
p c8c0487ab [Thomas Landauer @ 9 hours ago] Update transactions.rst
```

If that's also your case, please try getting to this state:

```
p 3d2f79f2d [Thomas Landauer @ 27 hours ago] Removing paragraph about PDO similarity
f c3721d990 [Thomas Landauer @ 23 hours ago] Update README.md
f 2de7b5b5a [Thomas Landauer @ 23 hours ago] Update README.md
f 8a5d0173e [Thomas Landauer @ 23 hours ago] Update introduction.rst
f 6cf002580 [Thomas Landauer @ 23 hours ago] Update data-retrieval-and-manipulation.rst
f a6edade26 [Thomas Landauer @ 23 hours ago] Removing "This is a Doctrine 2.1 feature"
f c54601f54 [Thomas Landauer @ 23 hours ago] Update query-builder.rst
f 4732b690e [Thomas Landauer @ 23 hours ago] Update security.rst
f 363fbd6c5 [Thomas Landauer @ 23 hours ago] Update transactions.rst
f 5226d0112 [Thomas Landauer @ 22 hours ago] Update transactions.rst
f 3ca8ba648 [Thomas Landauer @ 9 hours ago] Update introduction.rst
f 9b6ac8f50 [Thomas Landauer @ 9 hours ago] Update security.rst
f c8c0487ab [Thomas Landauer @ 9 hours ago] Update transactions.rst
```

I just did and there were no conflicts.

If you want to learn more about `git rebase`, please watch https://www.youtube.com/watch?v=uI1V7771plw&t=814s

ThomasLandauer commented:
Thanks! I think it worked out this time. Last time, the nano file contained hundreds (thousands?) of lines; this time only ~10.

greg0ire commented:
I think you may want to change your commit message now (with `git commit --amend`) since your changes are more than just about removing a paragraph.

ThomasLandauer commented:
OK, thanks! :-)
